handlers.js

```diff
@@ -318,14 +508,330 @@
+                                                finished.bind(this)(allresults)
+                                            } else {
+                                                allresults = allresults.concat( results )
+                                                reader.readEntries( onreadsuccess.bind(this),
+                                                                    onreaderr.bind(this) )
+                                            }
+                                        }
+
+                                        //console.log('readentries')
+                                        reader.readEntries( onreadsuccess.bind(this),
+                                                            onreaderr.bind(this))
+
+
+                                        } else {
+                                            excludedothtmlcheck.bind(this)()
+                                        }
+                                    } else {
+                                        excludedothtmlcheck.bind(this)()
+                                    }
+                                    }
+                                    if (auth && authdata.type == 401) {
+                                            var validAuth = false
+                                            var auth = this.request.headers['authorization']
+                                            if (auth) {
+                                                if (auth.slice(0,6).toLowerCase() == 'basic ') {
+                                                    var userpass = atob(auth.slice(6,auth.length)).split(':')
+                                                    if (userpass[0] == authdata.username && userpass[1] == authdata.password) {
+                                                        validAuth = true
+                                                    }
+                                                }
+                                            }
+                                            if (! validAuth) {
+                                                if (this.app.opts.optCustom401) {
+                                                    this.fs.getByPath(this.app.opts.optCustom401location, (file) => {
+                                                    if (! file.error) {
+                                                        file.file( function(filee) {
+                                                            var reader = new FileReader();
+                                                            reader.onload = function(e){
+                                                                this.useDefaultMime = false
+                                                                var data = e.target.result
+                                                                this.setHeader('content-type','text/html; charset=utf-8')
+                                                                this.setHeader("WWW-Authenticate", "Basic")
```

Comment:
Cool I didn't know you could add basic auth in htaccess.
It might be also a good idea to just have a global basic auth option in the settings (without having to learn the htaccess stuff). e.g. "Use basic auth" in advanced settings. Very cool!

Reply:
Yeah! I was doing some digging in the webapp.js and I found that this was already implemented! We just need to do the UI. (I am not marking this as resolved. I am going to do this real quick)